CRM/Contact/BAO/Query.php

```diff
@@ -1020,11 +1020,27 @@ public function addHierarchicalElements() {
     $addressCustomFields = CRM_Core_BAO_CustomField::getFieldsForImport('Address');
     $addressCustomFieldIds = array();
 
-    foreach ($this->_returnProperties['location'] as $name => $elements) {
+    foreach ($this->_returnProperties['location'] as $locationKey => $elements) {
+      if (is_numeric($locationKey) && !in_array($locationKey, $locationTypes)) {
```

Comment:
in_array($locationKey, $locationTypes) is on the off change that name is a number